monai/data/nifti_reader.py

```diff
@@ -135,3 +135,38 @@ def __getitem__(self, index):
                 continue
             compatible_meta[meta_key] = meta_datum
         return img, target, compatible_meta
+
+
+@export("monai.data")
+class Datasetd(Dataset):
+    """
+    General Dataset to handle dictionary format data, it can operate transforms for specific fields.
+    Input data should be a list of dictionaries, for example:
+    [{                            {                            {
+         'img': 'image1.nii.gz',      'img': 'image2.nii.gz',      'img': 'image3.nii.gz',
+         'seg': 'label1.nii.gz',      'seg': 'label2.nii.gz',      'seg': 'label3.nii.gz',
+         'extra': 123,                'extra': 456,                'extra': 789,
+         'shape': 'CHWD'              'shape': 'CHWD'              'shape': 'CHWD'
+     },                           },                           }]
+    """
+
+    def __init__(self, data, transform=None):
+        """
+        Args:
+            data (dict): input data to load and transform to generate dataset for model.
+            transform (Callable, optional): dict transforms to excute operations on dictionary data.
+        """
+        assert isinstance(data, list) and all(isinstance(item, dict) for item in data), \
```

Comment:
I think the pythonic way here would be not restricting the types, as long as `data[index]` returns some data, there's no need to make type assumption. For example `data` could even be a `torch.utils.data.dataset` object, as long as `self.data[index]` returns something we don't raise an error at this level.

Reply:
Updated in the latest patch.
Thanks.